docker/fetch.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Download as many external dependancies as possible to minimise
+# the amount of repeated downloads when building Docker images
+#
+# The downloaded files are stored in the 'dl' directory in the
+# top of the AliceVision source, and copied into the Docker enviroments
+# when the images are built
+
+set -e
+
+test -e docker/fetch.sh || {
+	echo This script must be run from the top level of the AliceVision tree
+	exit 1
+}
+
+test -d dl || \
+        mkdir dl
+test -f dl/vlfeat_K80L3.SIFT.tree || \
+        wget https://gitlab.com/alicevision/trainedVocabularyTreeData/raw/master/vlfeat_K80L3.SIFT.tree -O dl/vlfeat_K80L3.SIFT.tree
+test -f dl/cmake-3.16.3.tar.gz || \
+        wget https://cmake.org/files/v3.16/cmake-3.16.3.tar.gz -O dl/cmake-3.16.3.tar.gz
+test -d  dl/deps || \
+	mkdir dl/deps
+
+test -d build-fetch || {
+	mkdir -p build-fetch/external
+	ln -s ../../dl/deps build-fetch/external/download
+}
+
+pushd build-fetch
+cmake .. \
```

Comment:
This requires to have a valid compilation environment on the host side.

Reply:
I dont see an alternative, so I will merge it like that.

Reply:
There are a couple of other options, but they are terrible in their own ways:
1. Hardcode the URLs to fetch in the script, which will require version bumps to be done in 2 places
2. Try and parse CMakeLists.txt in fetch.sh, which would add a lot of complexity

Reply:
As an improvement for later - we could have a 'fetch' docker image which pulls in a volume, downloads the stuff and terminates